client/src/js/components/bhDischargeSelect/bhDischargeSelect.js

```diff
@@ -0,0 +1,28 @@
+angular.module('bhima.components')
+  .component('bhDischargeSelect', {
+    bindings : {
+      onChangeCallback : '&',
```

Comment:
It looks like you need to complete the second part of this - I feel that it might replace the old value with the first one in the dechargeTypes list without it.